texar/modules/pretrained/xlnet_utils.py

```diff
@@ -0,0 +1,247 @@
+            assign(linear.bias, prefix + "bias")
+
+    def assign_layer_norm(layer_norm: nn.LayerNorm, prefix: str):
+        assign(layer_norm.weight, prefix + "LayerNorm/gamma")
+        assign(layer_norm.bias, prefix + "LayerNorm/beta")
+
+    def load_xlnet_model(xlnet):
+        n_layers = len(xlnet.attn_layers)
+        for bias_name in ['r_r_bias', 'r_w_bias', 'r_s_bias']:
+            weight = get_weight("transformer/" + bias_name)
+            if xlnet.hparams.untie_r:
+                for idx in range(n_layers):
+                    layer: RelativeMultiheadAttention = xlnet.attn_layers[idx]
+                    assign(getattr(layer, bias_name), weight[idx])
+            else:
+                assign(getattr(xlnet, bias_name), weight)
+        assign(xlnet.word_embed.weight,
+               "transformer/word_embedding/lookup_table")
+
+        for idx in range(n_layers):
+            layer: RelativeMultiheadAttention = xlnet.attn_layers[idx]
+            prefix = f"transformer/layer_{idx}/rel_attn/"
+            qkv_weights = [get_weight(prefix + f"{part}/kernel")
+                           for part in "qkv"]
+            assign(layer.head_projection.weight,
+                   torch.cat([
+                       p.view(p.size(0), -1) for p in qkv_weights
+                   ], dim=1).t())
+            assign_linear(layer.pos_projection, prefix + "r/")
+            assign(layer.output_projection.weight,  # DO NOT TRANSPOSE THIS!!!!
+                   prefix + "o/kernel", lambda p: p.view(p.size(0), -1))
+            assign_layer_norm(layer.layer_norm, prefix)
+
+        for idx in range(n_layers):
+            layer: PositionWiseFF = xlnet.ff_layers[idx]
+            prefix = f"transformer/layer_{idx}/ff/"
+            for linear_idx in range(1, 2 + 1):
+                linear_prefix = f"{prefix}layer_{linear_idx}/"
+                linear_layer: nn.Linear = getattr(layer, f"linear{linear_idx}")
+                assign_linear(linear_layer, linear_prefix)
+            assign_layer_norm(layer.layer_norm, prefix)
+
+        seg_embeds = [
+            p.squeeze(0)
+            for p in torch.chunk(
+                get_weight("transformer/seg_embed"), n_layers, dim=0)]
+        for idx in range(n_layers):
+            assign(xlnet.attn_layers[idx].segment_embed, seg_embeds[idx])
+
+        if hasattr(xlnet, 'mask_emb') and hasattr(xlnet, 'lm_bias'):
+            assign(xlnet.mask_emb, "transformer/mask_emb/mask_emb")
+            assign(xlnet.lm_bias, "lm_loss/bias")
+
+    load_xlnet_model(model)
+
+    if len(from_params) > 0:
+        print(f"WARNING: Certain weights from checkpoint are not loaded: "
+              f"{list(from_params.keys())}")
+
+    filtered_to_params = [k for k in to_params if k.startswith("xlnet")]
+    if len(filtered_to_params) > 0:
+        print(f"WARNING: Certain parameters are not initialized: "
+              f"{list(filtered_to_params)}")
+
+
+def _default_download_dir() -> str:
```

Comment:
This same method is written in every `*_utils` file. Please extract them into another file (e.g. `pretrained_utils.py`) and reuse the code.